pkg/manager/internal.go

```diff
@@ -473,8 +473,8 @@ func (cm *controllerManager) Start(ctx context.Context) (err error) {
 			}
 		} else {
 			// Treat not having leader election enabled the same as being elected.
+			cm.startLeaderElectionRunnables()
 			close(cm.elected)
```

Comment:
@vincepri this is consistent today. We first close the channel, the we start the Runnables.

Also, is it intended to make the `cm.StartLeaderElectionRunnables` synchronous?

Reply:
Yes, like it's here

Reply:
ah now I get it, this is abusing the fact that `startLeaderElectionRunnables` synchronously waits for the cache to be synced. A tad implicit and has a good chance to be accidentally broken in the future again, but seems reasonable

Reply:
Now leader and leaderless starting of pre-election runnables is the same, that makes sense to me.